isobus/include/isobus/isobus/can_network_manager.hpp

```diff
@@ -133,18 +145,24 @@ namespace isobus
 		friend class CANLibProtocol;
 
 		/// @brief Adds a PGN callback for a protocol class
+		/// @details You may pass a destination `InternalControlFunction`, which will filter for only those messages
+		/// that target this source/destination pair (see https://github.com/ad3154/Isobus-plus-plus/issues/206).
 		/// @param[in] parameterGroupNumber The PGN to register for
 		/// @param[in] callback The callback to call when the PGN is received
 		/// @param[in] parentPointer A generic context variable that helps identify what object the callback was destined for
+		/// @param[in] destinationFunction An optional internal function destination to filter messages by
 		/// @returns `true` if the callback was added, otherwise `false`
-		bool add_protocol_parameter_group_number_callback(std::uint32_t parameterGroupNumber, CANLibCallback callback, void *parentPointer);
+		bool add_protocol_parameter_group_number_callback(std::uint32_t parameterGroupNumber, CANLibCallback callback, void *parentPointer, InternalControlFunction *destinationFunction = nullptr);
```

Comment:
Probably don't need it for the protocols? They bypass a lot of checks that the partner control function callbacks do and manage multiple internal control functions in the same callback anyways already.